File: src/orchestrator/internal/storage/local.go
package storage

import (
    "net/http"
    "os"
    "path/filepath"

    "github.com/gin-gonic/gin"
)

type LocalStorage struct {
    BasePath  string
    PublicURL string
}

func (l *LocalStorage) SaveFile(filename string, data []byte) (string, error) {
    fullPath := filepath.Join(l.BasePath, filename)
    err := os.WriteFile(fullPath, data, 0644)
    return fullPath, err
}

func (l *LocalStorage) GetFile(path string) ([]byte, error) {
    fullPath := filepath.Join(l.BasePath, path)
    return os.ReadFile(fullPath)
}

func (s *LocalStorage) HTTPHandler() gin.HandlerFunc {
    fs := gin.Dir(s.BasePath, false)
    return gin.WrapH(http.StripPrefix("/files/", http.FileServer(fs)))
}

func (l *LocalStorage) GetURL(path string) string {
    return l.PublicURL + "/" + path
}

File: src/phrase-service/storage/local.py
from .base import Storage
import os
from config import settings

class LocalStorage(Storage):
    def __init__(self, base_path=settings.AUDIO_DIR, public_url=settings.FILES_URL):
        self.base_path = base_path
        self.public_url = public_url

    def save_file(self, filename, data):
        full_path = os.path.join(self.base_path, filename)
        with open(full_path, "wb") as f:
            f.write(data)
        return full_path

    def get_url(self, path):
        return f"{self.public_url}/{path}"

